config/auth_server.go

```diff
@@ -0,0 +1,161 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// timeout to wait for local auth server to respond
+const authServerTimeout = 10 * time.Second
+
+type serverResponse struct {
```

Comment:
you should use `oauth2.Token` type for this one. it's already well-intergrated.

Reply:
it'd consider it a blocker if we don't return `oauth2.Token`-compatible response, because that's what we already have integrations in the other languages.